packages/api-server/api_server/routes/alerts.py

```diff
@@ -4,47 +4,107 @@
 from rx import operators as rxops
 
 from api_server.fast_io import FastIORouter, SubscriptionRequest
-from api_server.models import tortoise_models as ttm
+from api_server.gateway import rmf_gateway
+from api_server.models import AlertRequest, AlertResponse
 from api_server.repositories import AlertRepository
 from api_server.rmf_io import alert_events
 
 router = FastIORouter(tags=["Alerts"])
 
 
-@router.sub("", response_model=ttm.AlertPydantic)
+@router.sub("/requests", response_model=AlertRequest)
 async def sub_alerts(_req: SubscriptionRequest):
-    return alert_events.alerts.pipe(rxops.filter(lambda x: x is not None))
+    return alert_events.alert_requests.pipe(rxops.filter(lambda x: x is not None))
 
 
-@router.get("", response_model=List[ttm.AlertPydantic])
-async def get_alerts(repo: AlertRepository = Depends(AlertRepository)):
-    return await repo.get_all_alerts()
+@router.post("/request", status_code=201, response_model=AlertRequest)
+async def create_new_alert(
+    alert: AlertRequest, repo: AlertRepository = Depends(AlertRepository)
+):
+    """
+    Creates a new alert.
+    """
+    created_alert = await repo.create_new_alert(alert)
+    if created_alert is None:
+        raise HTTPException(409, f"Failed to create alert with ID {alert.id}")
```

Comment:
> Normally you wouldn't let the client choose the id. Is there a use case for this behavior?

Yeah I agree to some extent. But there is no clean way to perform ID cross checking  between the SCAS and downstream FA, if the comms with the FA is only one-way (ROS 2 pub). With the FA creating the ID, it will at least have the ability to subscribe and find the response for its own alert that it created.

> Catch `tortoise.exceptions.IntegrityError`.

https://github.com/open-rmf/rmf-web/commit/c5dc60b9c5738ebd628561529be380969f1db0f6